### PR TITLE
Update gapic/ to the generated client version.

### PR DESCRIPTION
**Pull request by shinfan**

### CODE REVIEW

**Comment by garrettjonesgoogle** [comment]
@michaelbausor could you do a pass first?

File: src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java [hunk 1]
```diff
@@ -40,6 +40,12 @@ public boolean hasVersion() {
     return version() != null && version().length() > 0;
   }
 
+  public abstract String generatorVersion();
+
+  public boolean hasGeneratorVersion() {
+    return generatorVersion() != null && generatorVersion().length() > 0;
```
Comment (bjwatson) [inline src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java]: It's slightly more efficient to cache the value of `generatorVersion()` and then do these tests.

File: src/main/resources/com/google/api/codegen/java/settings.snip [hunk 2]
```diff
@@ -4,7 +4,11 @@
   {@renderFileHeader(settingsFile.fileHeader)}
 
   {@settingsDoc(settingsFile.settings.doc)}
-  @@Generated("by GAPIC")
+  @if settingsFile.fileHeader.hasGeneratorVersion
+    @@Generated("by GAPIC v{@settingsFile.fileHeader.generatorVersion}")
```
Comment (bjwatson) [inline src/main/resources/com/google/api/codegen/java/settings.snip]: reverse the `{` and `@`

Comment (garrettjonesgoogle) [inline src/main/resources/com/google/api/codegen/java/settings.snip]: @bjwatson that looks wrong...

File: src/test/java/com/google/api/codegen/testdata/java_settings_library.baseline [hunk 3]
```diff
@@ -125,7 +125,7 @@ import org.joda.time.Duration;
  * </code>
  * </pre>
  */
-@Generated("by GAPIC")
+@Generated("by GAPIC v{settingsFile.fileHeader.generatorVersion}")
```
Comment (bjwatson) [inline src/test/java/com/google/api/codegen/testdata/java_settings_library.baseline]: @garrettjonesgoogle I was attempting to debug this.

Comment (shinfan) [inline src/test/java/com/google/api/codegen/testdata/java_settings_library.baseline]: My bad, I forgot to update the baseline after making the change. Now this should be resolved.

File: src/main/resources/com/google/api/codegen/java/settings.snip (hunk 2 above)
Comment (shinfan) [inline src/main/resources/com/google/api/codegen/java/settings.snip]: Baseline updated. Done.

File: src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java (hunk 1 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java]: Done

**Comment by shinfan** [comment]
@bjwatson PTAL

File: src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java (hunk 1 above)
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java]: I don't think we should add the complexity of caching. We don't have a performance problem in the generator. I'd like to revert this caching.

Comment (bjwatson) [inline src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java]: I wasn't asking for anything that complicated. All I meant was:

```java
public boolean hasGeneratorVersion() {
    String version = generatorVersion();
    return version != null && version.length() > 0;
}
```

Sorry for the ambiguity.

**Comment by michaelbausor** [comment]
LGTM, wait on @bjwatson before merging

File: src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java (hunk 1 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java]: Oop, misunderstood the meaning of 'cache'. Done.

**Pull request merged**